src/Knp/Component/Pager/Event/Subscriber/Sortable/ArraySubscriber.php

```diff
@@ -2,82 +2,93 @@
 
 namespace Knp\Component\Pager\Event\Subscriber\Sortable;
 
-use \ReflectionClass;
-use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Knp\Component\Pager\Event\ItemsEvent;
-use Doctrine\ORM\Query;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
 
 class ArraySubscriber implements EventSubscriberInterface
 {
-
     /**
      * @var string the field used to sort current object array list
      */
-    private $currentSortingFieldGetter;
+    private $currentSortingField;
 
     /**
      * @var string the sorting direction
      */
     private $sortDirection;
 
-    public function items(ItemsEvent $event)
+    /**
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    public function __construct(PropertyAccessorInterface $accessor = null)
     {
-        if (is_array($event->target) && count($event->target) > 1)
```

Comment:
What do you mean? Like first PR purely to refactor and second PR with actual feature pointed into first branch/PR? Sure can do so if you prefer that, even though I prefer having it in all in single PR to prevent conflicts (if there are some requested changes in first PR, it's likely second PR will have to be changed as well).